### PR TITLE
dmic: allow to set IPM_SOURCE_MODE from config blob

### PR DESCRIPTION
**Pull request by tmleman**
This bit determines operation mode for all source decimator.
0 = MONO
1 = STEREO

Signed-off-by: Tomasz Leman <tomasz.m.leman@intel.com>

### CODE REVIEW

File: src/include/sof/drivers/dmic.h [hunk 1]
```diff
@@ -250,6 +252,7 @@
 #define OUTCONTROL1_IPM_SOURCE_2(x)		SET_BITS(12, 11, x)
 #define OUTCONTROL1_IPM_SOURCE_3(x)		SET_BITS(10, 9, x)
 #define OUTCONTROL1_IPM_SOURCE_4(x)		SET_BITS(8, 7, x)
+#define OUTCONTROL1_IPM_SOURCE_MODE(x)	SET_BIT(6, x)
```
Comment (lgirdwood) [inline src/include/sof/drivers/dmic.h:255]: Can you check here alignment is good (and above) as GH sometimes does not render it correctly for review.

**Review by lgirdwood — COMMENTED** [review]
@singalsu any comments ?

File: src/include/sof/drivers/dmic.h (hunk 1 above)
Comment (tmleman) [reply src/include/sof/drivers/dmic.h:255]: When setting the tab to a size of 4 spaces is aligned with the line above.

Comment (lgirdwood) [reply src/include/sof/drivers/dmic.h:255]: ah, ok - we are using tab size 8. Fwiw, SOF uses the same coding style as Linux and Zephyr. See https://docs.zephyrproject.org/latest/contribute/index.html

**Comment by lgirdwood** [comment]
SOFCI TEST

**Pull request merged**